docs/Explanation/ADR/adr-004-did.md

```diff
@@ -0,0 +1,687 @@
+            "type": "EcdsaSecp256k1RecoveryMethod2020",
+            "controller": "did:key:cash1ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj",
+            "blockchainAccountID": "cash1ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj"
+        }
+    ],
+    "authentication": [
+        "did:key:cash1ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj"
+    ]
+}
+
+/* DID metadata */
+{
+  "created": "2021-03-23T06:35:22Z",
+  "updated": "2021-03-23T06:35:22Z", 
+  "versionId": "ae11692325525e82337167fcfab34d45d1904ff786e2d4bf4be2d1c4878cd34c" /* hex(blake2b(tx)) */
+}
+```
+
+Note that the representation is not compatible with the JSON-LD standard due to some specificity of the Protobuf message format.
+
+###### Update
+
+There are two ways of updating a DID document:
+
+- manage DID controllers
+- manipulate verification methods and relationships 
+
+In both cases the target DID MUST exists on chain and the `signerAccount` MUST exists as a verification method (property `blockchainAccountID`) in a verification relationship of type `authentication` or being listed as a DID controller.
+
+
+**DID controllers** 
+
+The DID controllers MAY be set using the gRPC message:
+
+`MsgUpdateDidDocument(did string, controllers []string, signerAccount string)` 
+
+The parameter `did` identifies the did document, the `controllers` are a list of DIDs that will replace the DID document controllers list and the `signerAccount` is the account address signing the transaction.
+
+Controllers will be added using the `did:keys` method.
+
+
+Example:
+
+```javascript
+/* gRPC message */
+MsgUpdateDidDocument(
+    "did:cosmos:cash:806e557e-ecdb-4e80-ab0d-a82ad35c9ceb", 
+    ["cosmos195rlq2hjnn2tmagskys4xtsnsey6gjljg8zxrn"],
+    "cash1ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj"
+)
+
+/* DID document */
+{
+    "context": [
+        "https://www.w3.org/ns/did/v1"
+    ],
+    "id": "did:cosmos:cash:806e557e-ecdb-4e80-ab0d-a82ad35c9ceb",
+    "controller": [
+        "did:key:cosmos195rlq2hjnn2tmagskys4xtsnsey6gjljg8zxrn"  // <-- new controller added
+    ],
+    "verificationMethods" : [
+        {
+            "id": "did:key:cash1ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj#cash1ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj",
+            "type": "EcdsaSecp256k1RecoveryMethod2020",
+            "controller": "did:key:cash1ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj",
+            "blockchainAccountID": "cash1ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj"
+        }
+    ],
+    "authentication": [
+        "did:key:cash1ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj#cash1ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj"
+    ]
+}
+
+/* DID metadata */
+{
+  "created": "2021-03-23T06:35:22Z",
+  "updated": "2021-04-23T06:35:22Z",  // <--  update field modified
+  "versionId": "96b3504be7e37a6aa55faff3cd41266bf4db3b0654263e1d9b779d3b30174dd1", /* hex(blake2b(tx)) */  //<-- new hash computed
+  "deactivated": false
+}
+
+```
+
+
+**Verification methods and relationships**
```

Comment:
```suggestion
**Manipulate Verification Methods and Relationships**
```

Reply:
oh sorry I didn't notice until now, wdyt about using Markdown heading hierarchical syntax instead of just bold?
so instead of **Bold Heading** we could use ### Bold Heading (or whatever level you like)

using heading levels is nifty for generating TOCs too, we can limit the levels to show in TOC if TOC length is a concern

Reply:
I need to review the source again why I didn't use the ### here, but it was on purposes